src/acquisition/covidcast/csv_to_database.py

```diff
@@ -73,27 +73,21 @@ def archive_as_successful(path_src, filename, source):
 
     (source, signal, time_type, geo_type, time_value) = details
 
-    # iterate over rows and upload to the database
     all_rows_valid = True
-    for row_values in csv_importer_impl.load_csv(path, geo_type):
-      if not row_values:
-        all_rows_valid = False
-        continue
-
-      try:
-        database.insert_or_update(
-            source,
-            signal,
-            time_type,
-            geo_type,
-            time_value,
-            row_values.geo_value,
-            row_values.value,
-            row_values.stderr,
-            row_values.sample_size)
-      except Exception as e:
+    csv_rows = csv_importer_impl.load_csv(path, geo_type)
+
+    if not csv_rows:
```

Comment:
`csv_rows` is a generator and will always be truthy; you probably want `any()` or `all()` here instead

Reply:
good catch